### ISSUE TITLE
SD 2.1 使用one_flow编译后，输出图片变黑了

### ISSUE THREAD

**Issue by chenly15** [siliconflow/onediff#722]
#### OneDiff git commit id
a09dc392250f450eec8194bbcc8078b7e20b5f54

####  OneFlow version info
Run `python -m oneflow --doctor` and paste it here.
version: 0.9.1.dev20240312+cu118
git_commit: 564fc15
cmake_build_type: Release
rdma: True
mlir: True
enterprise: False

### How To Reproduce
Steps to reproduce the behavior(code or script):
import torch
from diffusers import StableDiffusionPipeline, DPMSolverMultistepScheduler

model_id = "stabilityai/stable-diffusion-2-1"

pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16, local_files_only=True)
pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config)
pipe = pipe.to("cuda")

from onediff.infer_compiler import oneflow_compile
pipe.unet = oneflow_compile(pipe.unet)
pipe.vae.decoder = oneflow_compile(pipe.vae.decoder)

prompt = "a photo of an astronaut riding a horse on mars"
image = pipe(prompt).images[0]
    
image.save("astronaut_rides_horse.png")

### The complete error message
无报错信息

### Additional context
None.

**Comment by strint** [comment]
@lixiang007666

**Comment by lixiang007666** [comment]
Refer to this PR to modify your script: https://github.com/siliconflow/onediff/pull/725.
@chenly15

**Comment by chenly15** [comment]
> import torch from diffusers import StableDiffusionPipeline, DPMSolverMultistepScheduler
> 
> model_id = "stabilityai/stable-diffusion-2-1"
> 
> pipe = StableDiffusionPipeline.from_pretrained(model_id, torch_dtype=torch.float16, local_files_only=True) pipe.scheduler = DPMSolverMultistepScheduler.from_config(pipe.scheduler.config) pipe = pipe.to("cuda")
> 
> from onediff.infer_compiler import oneflow_compile pipe.unet = oneflow_compile(pipe.unet) pipe.vae.decoder = oneflow_compile(pipe.vae.decoder)
> 
> prompt = "a photo of an astronaut riding a horse on mars" image = pipe(prompt).images[0]
> 
> image.save("astronaut_rides_horse.png")

Thanks. It works after setting the environment variable. How does it effect the result? When and how I to set it up in other stable diffusion models?

**Comment by chenly15** [comment]
> Refer to this PR to modify your script: #725. @chenly15



> Refer to this PR to modify your script: ##718

**Comment by chenly15** [comment]
> Refer to this PR to modify your script: #725. @chenly15

Thanks. It works after setting the environment variable. How does it effect the result? When and how I to set it up in other stable diffusion models? @lixiang007666

**Comment by lixiang007666** [comment]
When overflow issues occur (black images), you can resolve them by setting this environment variable. 

The reason is that using FP16 accumulation can easily lead to overflow issues in MHA. However, completely disabling half-precision accumulation g in MHA (setting `ONEFLOW_ATTENTION_ALLOW_HALF_PRECISION_ACCUMULATION` to `False`) would slow down inference speed. 

The environment variable `ONEFLOW_ATTENTION_ALLOW_HALF_PRECISION_SCORE_ACCUMULATION_MAX_M` allows for control over the use of half-precision accumulation. Setting it to `0` or a positive value permits limited use of half-precision accumulation. This ensures correct output while achieving optimal inference time.

@chenly15

**Comment by strint** [comment]
Fixed with: https://github.com/siliconflow/onediff/pull/725
